### PR TITLE
feat(payments): PAYPAL-7 Pass in merchant ID on Pa…

### PR DESCRIPTION
**Pull request by bc-fetisov**
…yPal button script for PayPal Express Checkout

## What?
Add possibility to add script data attributes

## Why?
The PayPal Shopping feature should know which merchants to target for PayPal Store Cash

## Testing / Proof
Screen Shot 2019-10-18 at 14 27 22
Screen Shot 2019-10-18 at 14 28 36

Screen Shot 2019-10-18 at 14 28 21

@bigcommerce/frontend

### CODE REVIEW

File: src/script-loader.ts [hunk 1]
```diff
@@ -22,12 +26,18 @@ export default class ScriptLoader {
         private _requestSender: RequestSender
     ) {}
 
-    loadScript(src: string, options?: LoadScriptOptions): Promise<void> {
+    loadScript(src: string, options?: LoadScriptOptions, scriptAttributes?: ScriptAttributes): Promise<void> {
```
Comment (davidchin) [inline src/script-loader.ts]: Just add the additional attributes as a part of `LoadScriptOptions` instead of introducing a third parameter.

Comment (bc-fetisov) [reply src/script-loader.ts]: Done @davidchin

**Comment by BC-ACherednichenko** [comment]
ping @davidchin

File: src/script-loader.ts [hunk 2]
```diff
@@ -27,6 +32,13 @@ export default class ScriptLoader {
             this._scripts[src] = new Promise((resolve, reject) => {
                 const script = document.createElement('script') as LegacyHTMLScriptElement;
                 const { async = false } = options || {};
+                const scriptAttributes = options && options.attributes;
```
Comment (davidchin) [inline src/script-loader.ts]: You can combine this with the line above. i.e.: 

```ts
const { async = false, attributes = {} } = options || {};
```

File: src/script-loader.ts [hunk 3]
```diff
@@ -27,6 +32,13 @@ export default class ScriptLoader {
...
+
+                for (const key in scriptAttributes) {
+                    if (scriptAttributes.hasOwnProperty(key)) {
```
Comment (davidchin) [inline src/script-loader.ts]: You can just use `Object.keys` so you don't need to do this check. i.e.:

```ts
Object.keys(attributes)
    .forEach(key => {
        script.setAttribute(key, attributes[key]);
    });
```

Comment (bc-fetisov) [reply src/script-loader.ts]: Fixed

**Pull request merged**